shared/src/main/java/com/splunk/opentelemetry/javaagent/shared/MiddlewareHolder.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry.javaagent.shared;
+
+import java.util.concurrent.atomic.AtomicReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MiddlewareHolder {
+
+  private static final Logger log = LoggerFactory.getLogger(MiddlewareHolder.class);
+
+  public static final AtomicReference<String> middlewareName = new AtomicReference<>();
+  public static final AtomicReference<String> middlewareVersion = new AtomicReference<>();
+
+  public static void trySetName(String name) {
+    if (!middlewareName.compareAndSet(null, name)) {
+      log.debug("Trying to re-set middleware name from {} to {}", middlewareName.get(), name);
```

Comment:
Will this spam logs?

Reply:
> Will this spam logs?
By current design - no. The idea is to set middleware attributes only once as soon as possible. This log entry should indicate that there are either two competing middleware detection instrumentations or instrumentation is not good enough and it tries to set the attribute multiple times. In the first case there is something to investigate, in the second - find a better place to instrument.

> Just to confirm: this is using the shaded logger, right?
How do I check? :)

Reply:
> How do I check? :)

* Unzip the javaagent jar
* Find the class file (it should not be a classdata file, that would mean that it won't be loaded by bootstrap classloader)
* `javap -c <class>`

I know that it's tedious, but since I'm not that good at gradle it's the quickest way for me to check for things like that.

Reply:
I'd say it's shaded:
```
  static {};
    Code:
       0: ldc           #2                  // class com/splunk/opentelemetry/javaagent/bootstrap/MiddlewareHolder
       2: invokestatic  #56                 // Method io/opentelemetry/javaagent/slf4j/LoggerFactory.getLogger:(Ljava/lang/Class;)Lio/opentelemetry/javaagent/slf4j/Logger;
```